backend/apps/moderation/views.py

```
@@ -0,0 +1,61 @@
+        qs = ContentReport.objects.all()
+        status_param = self.request.query_params.get("status", "PENDING")
+        if status_param and status_param != "ALL":
+            qs = qs.filter(status=status_param)
+        return qs
+
+    def perform_create(self, serializer):
+        try:
+            serializer.save(reporter=self.request.user)
+        except IntegrityError:
+            # Handle unique constraint violation
+            raise serializers.ValidationError({"error": "You have already reported this content."})
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)